virtool_cli/ref/otu.py

```diff
@@ -176,28 +305,71 @@ def add_sequences(
-            new_accessions=new_accessions,
+    return sorted(new_accessions)
+
+
+def _get_isolate_name(record: NCBIGenbank) -> IsolateName | None:
+    """Get the isolate name from a Genbank record"""
+    record_logger = logger.bind(
+        accession=record.accession,
+        definition=record.definition,
+        source_data=record.source,
+    )
+
+    if record.source.model_fields_set.intersection(
+        {IsolateNameType.ISOLATE, IsolateNameType.STRAIN, IsolateNameType.CLONE},
+    ):
+        for source_type in IsolateNameType:
+            if source_type in record.source.model_fields_set:
+                return IsolateName(
+                    type=IsolateNameType(source_type),
+                    value=record.source.model_dump()[source_type],
+                )
+
+    elif record.refseq:
+        record_logger.debug(
+            "RefSeq record does not contain sufficient source data. "
+            + "Edit before inclusion."
+        )
+
+        return IsolateName(
+            type=IsolateNameType(IsolateNameType.REFSEQ),
+            value=record.accession,
         )
 
-    else:
-        otu_logger.info("No new sequences added to OTU")
+    record_logger.debug("Record does not contain sufficient source data for inclusion.")
 
+    return None
 
-def get_molecule_from_records(records: list[NCBIGenbank]) -> Molecule:
-    """Return relevant molecule metadata from one or more records"""
-    for record in records:
-        if record.refseq:
-            return Molecule(
-                strandedness=record.strandedness.value,
-                type=record.moltype.value,
-                topology=record.topology.value,
+
+def _get_segments_from_records(records: list[NCBIGenbank]) -> list[Segment]:
+    if len(records) == 1:
+        record = records[0]
+
+        if record.source.segment != "":
+            segment_name = record.source.segment
+        else:
+            segment_name = record.source.organism
+
+        return [Segment(name=segment_name, required=True, length=len(record.sequence))]
+
+    segments = []
+    for record in sort_records_by_accession(records):
+        if record.source.segment:
+            segments.append(
+                Segment(
+                    name=record.source.segment,
+                    required=True,
+                    length=len(record.sequence),
+                )
             )
+        else:
+            raise ValueError("No segment name found for multipartite OTU segment.")
 
-    return Molecule(
-        strandedness=records[0].strandedness.value,
-        type=records[0].moltype.value,
-        topology=records[0].topology.value,
-    )
+    return segments
+
+
+def sort_records_by_accession(records: list[NCBIGenbank]):
+    """Sort a list of records by accession alphanumerically"""
+    record_dict = {record.accession: record for record in records}
+    return [record_dict[accession] for accession in sorted(record_dict.keys())]
```

Comment:
Only used in one place. I would get rid of the function and use this one liner:
```suggestion
    return sorted(records, key=lambda record: record.accession)
```